athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/propertygraph/PropertyGraphHandler.java

```diff
@@ -0,0 +1,231 @@
+                    graphTraversal = graphTraversal.elementMap();
+
+                    for (final Field nextField : recordsRequest.getSchema().getFields()) {
+                        EdgeRowWriter.writeRowTemplate(builder, nextField);
+                    }
+
+                    break;
+            }
+        }
+
+        // log string equivalent of gremlin query
+        logger.info("readWithConstraint: enter - "
+                + GroovyTranslator.of("g").translate(graphTraversal.asAdmin().getBytecode()));
+
+        final GraphTraversal graphTraversalFinal1 = graphTraversal;
+        final GeneratedRowWriter rowWriter1 = builder.build();
+
+        while (graphTraversalFinal1.hasNext() && queryStatusChecker.isQueryRunning()) {
+            numRows++;
+
+            spiller.writeRows((final Block block, final int rowNum) -> {
+                final Map obj = (Map) graphTraversalFinal1.next();
+                return (rowWriter1.writeRow(block, rowNum, (Object) obj) ? 1 : 0);
+            });
+        }
+
+        logger.info("readWithConstraint: numRows[{}]", numRows);
+    }
+
+    /**
+     * Used to generate Gremlin Query part for Constraint Map
+     * 
+     * @param traversal Gremlin Traversal, traversal is updated based on constraints
+     *                  map
+     * @param hasMap    Constraint Hash Map
+     * 
+     * @return A Gremlin Query Part equivalent to Contraint.
+     */
+    public GraphTraversal getQueryPartForContraintsMap(GraphTraversal traversal,
+            final ReadRecordsRequest recordsRequest) 
+            {
+        final Map<String, ValueSet> hashMap = recordsRequest.getConstraints().getSummary();
+        if (hashMap.size() == 0) {
+            return traversal;
+        }
+
+        logger.info("readWithContraint: Constaints Map " + hashMap.toString());
+
+        final Set<String> setOfkeys = (Set<String>) (hashMap.keySet());
+        for (final String key : setOfkeys) {
+            if (hashMap.get(key) instanceof SortedRangeSet) {
+                final List<Range> ranges = ((SortedRangeSet) hashMap.get(key)).getOrderedRanges();
+
+                for (final Range range : ranges) {
+                    if (!range.getLow().isNullValue() && !range.getHigh().isNullValue()) {
+                        if (range.getLow().getValue().toString().equals(range.getHigh().getValue().toString())) {
+                            traversal = GremlinQueryPreProcessor.generateGremlinQueryPart(traversal, key,
+                                    range.getLow().getValue().toString(), range.getType(), range.getLow().getBound(),
+                                    GremlinQueryPreProcessor.Operator.EQUALTO);
+                            break;
+                        }
+                    }
+
+                    if (!range.getLow().isNullValue()) {
+                        logger.info("inside flattenConstraintMap: " + range.getType().toString()
```

Comment:
I'd remove this log line or move it to DEBUG

Reply:
removed